### PR TITLE
[SQL] Improve error messages for unsupported constructs

### LINKED ISSUE #2722: Actionable error messages in SQL compiler instead of "not yet implemented" errors
We have certain functions that we don't support in specific contexts, like using row_number() in contexts with windowed aggregates. The rank/lead/lag functions come up with such errors too.

We currently return an opaque "Not yet implemented" error for these. We've had consistent user feedback that this is confusing, when they see these functions in our documentation.

I suggest we give the user a more actionable error instead. Something like: "`function` not yet implemented in this specific context. Please file an issue here: https://github.com/feldera/feldera/issues".

Being more specific about context would also be great, but I understand that's really difficult.

### PR DESCRIPTION
**Pull request by mihaibudiu**
Fixes #2722

### CODE REVIEW

**Comment by mihaibudiu** [comment]
Fixes https://github.com/feldera/feldera/issues/2549

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java [hunk 1]
```diff
@@ -2250,7 +2260,7 @@ void visitSort(LogicalSort sort) {
             return;
         }
         if (sort.offset != null)
-            throw new UnimplementedException(node);
+            throw new UnimplementedException("OFFSET in SORT not yet implemented", node);
```
Comment (lalithsuresh) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java:2263]: The cases where we point to issue 457 are great. It's really clear that way and invites people to leave a comment to ask for support.

Can we add that pattern to some of the other cases (including this one, but also group-by + filter, join types, single aggregate + filter etc.)?

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/ir/expression/literal/DBSPVariantNullLiteral.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public boolean sameValue(@Nullable ISameValue o) {
 
     @Override
     public DBSPLiteral getWithNullable(boolean mayBeNull) {
-        throw new UnimplementedException("getWithNullable");
+        throw new InternalCompilerError("getWithNullable for VariantNullLiteral");
```
Comment (lalithsuresh) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/ir/expression/literal/DBSPVariantNullLiteral.java:40]: How does InternalCompilerError show up e2e (e.g., in the WebConsole). Does the error message ask users to file a bug?

**Review by lalithsuresh — APPROVED** [review]

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java (hunk 1 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java:2263]: Why add issues if no one asked for them? Issues are about prioritizing work.
We have an umbrella issue for unimplemented stuff: https://github.com/feldera/feldera/issues/172

Comment (lalithsuresh) [reply sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java:2263]: Lowers the barrier for someone to come in and leave a comment. Defaulting to #172 looks great actually.

**Comment by mihaibudiu** [comment]
I don't understand why this branch keeps failing in merge, I will delete it and push again

**Pull request closed without merging**